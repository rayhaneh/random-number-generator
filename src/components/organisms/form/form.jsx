import * as React from 'react';

import {
  Button,
} from '../../atoms';

import {
  InputFieldWithLabel
} from '../../molecules';

export const Form = (props) => {
  const {
    max,
    min,
    onMinChange,
    onMaxChange,
    onSubmit,
  } = props;

  return (
    <form 
      onSubmit={onSubmit}
      className={'pa5 flex flex-column'}
    >
      <div className={'flex flex-row mb2'}>
        <InputFieldWithLabel
          handleChange={onMinChange}
          label={'Min'}
          value={min}
          min={''}
          max={max}
        />
        <InputFieldWithLabel
          handleChange={onMaxChange}
          label={'Max'}
          value={max}
          min={min}
          max={''}
        />   
      </div>   
      <Button
        text={'GENERATE'}
      />
    </form>
  )
}
