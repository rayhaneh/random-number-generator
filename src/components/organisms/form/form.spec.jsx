import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import { Form } from './form';

let props;
let comp;

describe('Form Component', () => {
  beforeEach(() => {
    props = {
      max: 50,
      min: 0,
      onMaxChange: jest.fn(),
      onMinChange: jest.fn(),
      onSubmit: jest.fn(),
    };
  });

  it('renders properly', () => {
    comp = shallow(<Form {...props} />);
    expect(comp).toMatchSnapshot();
  });
});
