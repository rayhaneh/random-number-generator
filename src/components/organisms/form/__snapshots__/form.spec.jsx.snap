// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    max={50}
    min={0}
    onMaxChange={[MockFunction]}
    onMinChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex flex-row mb2"
        >
          <InputFieldWithLabel
            handleChange={[MockFunction]}
            label="Min"
            max={50}
            min=""
            value={0}
          />
          <InputFieldWithLabel
            handleChange={[MockFunction]}
            label="Max"
            max=""
            min={0}
            value={50}
          />
        </div>,
        <Button
          text="GENERATE"
        />,
      ],
      "className": "pa5 flex flex-column",
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <InputFieldWithLabel
              handleChange={[MockFunction]}
              label="Min"
              max={50}
              min=""
              value={0}
            />,
            <InputFieldWithLabel
              handleChange={[MockFunction]}
              label="Max"
              max=""
              min={0}
              value={50}
            />,
          ],
          "className": "flex flex-row mb2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleChange": [MockFunction],
              "label": "Min",
              "max": 50,
              "min": "",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleChange": [MockFunction],
              "label": "Max",
              "max": "",
              "min": 0,
              "value": 50,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "GENERATE",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex flex-row mb2"
          >
            <InputFieldWithLabel
              handleChange={[MockFunction]}
              label="Min"
              max={50}
              min=""
              value={0}
            />
            <InputFieldWithLabel
              handleChange={[MockFunction]}
              label="Max"
              max=""
              min={0}
              value={50}
            />
          </div>,
          <Button
            text="GENERATE"
          />,
        ],
        "className": "pa5 flex flex-column",
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <InputFieldWithLabel
                handleChange={[MockFunction]}
                label="Min"
                max={50}
                min=""
                value={0}
              />,
              <InputFieldWithLabel
                handleChange={[MockFunction]}
                label="Max"
                max=""
                min={0}
                value={50}
              />,
            ],
            "className": "flex flex-row mb2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [MockFunction],
                "label": "Min",
                "max": 50,
                "min": "",
                "value": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [MockFunction],
                "label": "Max",
                "max": "",
                "min": 0,
                "value": 50,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "GENERATE",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
